samcli/local/docker/container.py

```diff
@@ -201,6 +231,8 @@ def delete(self):
             # Skip such exceptions.
             if not removal_in_progress:
                 raise ex
+            # else debug log explaining that we had and skipped an exception
```

Comment:
I think we should actually move the two lines of comment from 230:231 here, they describe what we are doing.

Also, is passing the exception to the logger better that using its string representation? Like so:
`LOG.debug("Container removal is in progress, skipping exception", exc_info=True)`

Reply:
I just updated the comments above to explain the "else" condition